### PR TITLE
Fixes bug 1274345 - Added support from triming signatures missing symbols.

### PR DESCRIPTION
**Pull request by adngdb**

### CODE REVIEW

File: socorro/processor/signature_utilities.py [hunk 1]
```diff
@@ -243,17 +245,45 @@ def _do_generate(self,
...
+            # If the signature matches the irrelevant signatures regex,
+            # skip to the next frame.
             if self.irrelevant_signature_re.match(a_signature):
                 continue
+
+            # If the signature matches the trim dll signatures regex,
+            # rewrite it to remove all but the module name.
+            if self.trim_dll_signature_re.match(a_signature):
+                try:
+                    a_signature = a_signature[:a_signature.index('@')]
```
Comment (peterbe) [inline socorro/processor/signature_utilities.py]: Normally you do `a_signature = a_signature.split('@')[0]` which does the same thing but without having to compute the index first. And then you don't need the `try:except ValueError` blocks.

File: socorro/processor/signature_utilities.py [hunk 2]
```diff
@@ -243,17 +245,45 @@ def _do_generate(self,
...
+                except ValueError:
+                    # No '@' in that signature, we cannot trim it.
+                    pass
+
+                # If this trimmed DLL signature is the same as the previous
+                # frame's, we do not want to add it.
+                if (
+                    len(new_signature_list) and
```
Comment (peterbe) [inline socorro/processor/signature_utilities.py]: No need to use `len()`.

File: socorro/siglists/README.md [hunk 3]
```diff
@@ -31,6 +32,20 @@ File: [``prefix_signature_re.txt``](./prefix_signature_re.txt)
...
+Trim DLL Signatures are regular expressions of signatures that will be trimmed down to only their module name. For example, if the list contains ``foo32\.dll.*`` and a stack trace looks like this:
+
+<pre>0x0
+foo32.dll@0x2131
+foo32.dll@0x1943
+myFavoriteSig()
+</pre>
+
+The generated signature will be: ``0x0 | foo32.dll | myFavoriteSig()``.
+
```
Comment (peterbe) [inline socorro/siglists/README.md]: Great example! I appreciate this documentation.

**Comment by peterbe** [comment]
some nits.

File: socorro/processor/signature_utilities.py (hunk 2 above)
Comment (adngdb) [inline socorro/processor/signature_utilities.py]: I have been writing too much JavaScript lately, I forgot how good Python is... :)

**Comment by adngdb** [comment]
addressed

**Pull request merged**